### PR TITLE
feat(api): simplify parsing and validation of http input params

### LINKED ISSUE #3163: Simplify HTTP API input parameters check
Currently, a parameters check is performed at the beginning of each HTTP handler. This parameter check is very repetitive and unnecessarily increases the size of the handler's body. Find a way to reduce the repetitive code. Consider using the https://github.com/go-playground/validator library or any other library that accomplishes this task.

### PR DESCRIPTION
**Pull request by mrekucci**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Closes #3163 #3165 #3371

### Open API Spec Version Changes (if applicable)
Swarm.yaml: minor
SwarmDebug.yaml: minor
SwarmCommon.yaml: minor

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3344)

### CODE REVIEW

File: openapi/SwarmDebug.yaml [hunk 1]
```diff
@@ -92,8 +92,8 @@ paths:
             application/json:
```
Comment (aloknerurkar) [inline openapi/SwarmDebug.yaml:92]: I think changes to SwarmDebug.yaml can be avoided as we should be incrementing versions here when we plan to not support it going ahead.

File: pkg/api/balances.go [hunk 2]
```diff
@@ -33,12 +33,14 @@ type balancesResponse struct {
...
-func (s *Service) balancesHandler(w http.ResponseWriter, r *http.Request) {
+func (s *Service) balancesHandler(w http.ResponseWriter, _ *http.Request) {
+	logger := s.logger.WithName("get_consumed").Build()
+
 	balances, err := s.accounting.Balances()
 	if err != nil {
 		jsonhttp.InternalServerError(w, errCantBalances)
-		s.logger.Debug("balances: get balances failed", "error", err)
-		s.logger.Error(nil, "balances: get balances failed")
+		logger.Debug("balances: get balances failed", "error", err)
```
Comment (aloknerurkar) [inline pkg/api/balances.go:42]: remove "balances" from the msg

File: pkg/api/util.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+		return typ, err
+	},
+	"decBase64url": func(v string) (string, error) {
+		buf, err := base64.URLEncoding.DecodeString(v)
+		return string(buf), err
+	},
+}
+
+// mapStructure maps the input to the output values.
+// The input is on of the following:
```
Comment (aloknerurkar) [inline pkg/api/util.go:126]: one

File: pkg/api/api.go [hunk 4]
```diff
@@ -662,12 +752,12 @@ func equalASCIIFold(s, t string) bool {
 // direct push to the network (default) a pushStamperPutter is returned.
 // returns a function to wait on the errorgroup in case of a pushing stamper putter.
 func (s *Service) newStamperPutter(r *http.Request) (storage.Storer, func() error, error) {
-	batch, err := requestPostageBatchId(r)
+	batch, err := requestPostageBatchId(r) // TODO: extrapolate the headers parsing to the handler level!
```
Comment (notanatol) [inline pkg/api/api.go:755]: these `TODOs` are meant to be fixed in a future PR?

**Review by notanatol — APPROVED** [review]

File: pkg/api/welcome_message.go [hunk 5]
```diff
@@ -29,17 +29,19 @@ func (s *Service) getWelcomeMessageHandler(w http.ResponseWriter, r *http.Reques
 }
 
 func (s *Service) setWelcomeMessageHandler(w http.ResponseWriter, r *http.Request) {
+	logger := s.logger.WithName("post_wallet").Build()
```
[the remark this answers is not in the document]
Comment (mrekucci) [reply pkg/api/welcome_message.go]: Yup, fixed.

File: openapi/SwarmDebug.yaml (hunk 1 above)
Comment (mrekucci) [reply openapi/SwarmDebug.yaml:92]: Added the recomended semver changes to the PR description.

File: pkg/api/balances.go (hunk 2 above)
Comment (mrekucci) [reply pkg/api/balances.go]: Nice catch, done!

File: pkg/api/util.go (hunk 3 above)
Comment (mrekucci) [reply pkg/api/util.go]: Done.

File: pkg/api/api.go (hunk 4 above)
Comment (mrekucci) [reply pkg/api/api.go:755]: Yes, there is an existing issue for that: #3371

**Review by aloknerurkar — APPROVED** [review]
Looks really good!

**Comment by umerm-work** [comment]
Really nice !

File: pkg/log/log.go [hunk 6]
```diff
@@ -65,6 +65,16 @@ func ParseVerbosityLevel(s string) (Level, error) {
 	return Level(i), err
 }
 
+// MustParseVerbosityLevel returns a verbosity Level parsed from the given s.
+// It panics if the given s is not a valid verbosity level.
+func MustParseVerbosityLevel(s string) Level {
+	l, err := ParseVerbosityLevel(s)
+	if err != nil {
+		panic(err)
```
Comment (vladopajic) [inline pkg/log/log.go:73]: please handle error instead of panicing

Comment (mrekucci) [reply pkg/log/log.go:73]: This is the `Must` pattern commonly used in the std lib (e.g. `regexp.MustCompile`). It is used in cases where the input has already been validated and simplifies initialization.

Comment (vladopajic) [reply pkg/log/log.go:73]: Regardless of naming convention, having panic in code may lead to unexpected program termination. It is risky to be certain that some execution flows will not cause program to terminate.

Also, this rule is documented in [coding style](https://github.com/ethersphere/bee/blob/master/CODINGSTYLE.md#panic-with-care) doc.

**Pull request merged**